src/Functions/FunctionBase64Conversion.h

```diff
@@ -225,6 +218,16 @@ class FunctionBase64Conversion : public IFunction
         return dst_column;
     }
 };
+
+/** We must call it in advance from a single thread
+  *  to avoid thread sanitizer report about data race in "codec_choose" function.
+  */
+inline void initializeBase64()
```

Comment:
That's a kind of unbeautiful workaround. Yes, the extensive testing in ClickHouse's CI uncovers many sanitizer issues in submodules, and base64 seems no exception here. If possible, it would be appreciated if you could
- make a local tsan build of ClickHouse. For that, open the "BuilderDebTsan" log of an arbitrary other PR, e.g. [here](https://github.com/ClickHouse/ClickHouse/actions/runs/6042827088/job/16403984335?pr=54047), then search for "cmake " and run a stripped down version of this statement locally, e.g. `cmake  -DCMAKE_BUILD_TYPE=None -DSANITIZE=thread ..` + `ninja`
- try to reproduce the issue, e.g. run tests
- In the best case: fix the issue in base64 - we then can either reference a fork of "base64" in the ClickHouse [GItHub organization](https://github.com/ClickHouse/) with the fix or reference an upstream commit with the fix 
- use `__attribute__((no_sanitize("thread")))` (https://clang.llvm.org/docs/ThreadSanitizer.html), respectively the wrapper macro from "base/base/defines.h"

Reply:
Sure, will do!

Reply:
Also just for the record: I'm not sure this issue is still the case, as this comment was included back in the days when CH used `aklomp/base64` before `Turbo-Base64`.

Reply:
Feel free to make an experimental push to CI with the workaround removed. We can check if tsan (still) complains or not.